Documentation/supermuc.rst

```diff
@@ -0,0 +1,362 @@
+  export LD_LIBRARY_PATH=/dss/dsshome1/02/di52lak2/myLib/ASAGI/build/lib:$LD_LIBRARY_PATH
+
+
+3. Install libxsmm and ASAGI (Optional)
+
+Install libxsmm:
+
+::
+
+  $ git clone git@github:hfp/libxsmm.git
+  $ Cd libxsmm
+  $ make generator
+  (then add path-to-libxsmm/bin to PATH in bashrc)
+
+Install ASAGI:
+
+::
+
+  $ git clone git@github:TUM-I5/ASAGI.git
+  
+then copy fix_submodules.sh here in submodules/utils and run it to get all submodules cloned.
+
+set compiler options:
+
+::
+
+  $ export FC=mpif90
+  $ export CXX=mpiCC
+  $ export CC=mpicc
+
+  $ make build
+  $ Cd build
+  $ CMAKE_PREFIX_PATH=$NETCDF_BASE
+  $ cmake ../ -DSHARED_LIB=no -DSTATIC_LIB=yes -DNONUMA=on -DCMAKE_INSTALL_PREFIX=$HOME/<folder-to-ASAGI>/build/ 
+  $ make
+  $ make install
+  (Know errors: 1.Numa could not found - turn off Numa by -DNONUMA=on . )
+
+
+4. Copy the SeisSol configuration to a file e.g. supermuc_ng.py
+
+::
+
+  import os
+  # build options
+  compileMode                 = 'release'
+  #compileMode                 = 'relWithDebInfo'
+  #compileMode                 = 'debug'
+  parallelization             = 'hybrid'
+  #parallelization             = 'mpi'
+  generatedKernels            = 'yes'
+  #measureNodeLevelPerformance = 'none'
+  useExecutionEnvironment     = 'yes'
+  order = 4
+  equations='elastic'
+  #equations = 'viscoelastic2'
+  #numberOfMechanisms = 3
+  # machine dependent options
+  #compiler='gcc'
+  cppCompiler          = 'mpiCC'
+  fortranCompiler      = 'mpif90'
+
+  netcdf='yes'
+  hdf5='yes'
+  metis='yes'
+  netcdfDir=os.environ['NETCDF_BASE']
+  hdf5Dir=os.environ['HDF5_BASE']
+
+  # ASAGI folder need to be verified.
+  asagi='yes'
+  zlibDir='/dss/dsshome1/02/di52lak2/myLib/ASAGI/build/lib'
+
+  phase=3 # for Supermuc-NG
+  if phase==1:
+     arch ='dsnb'
+  elif phase==2:
+     arch = 'dhsw'
+     #commThread ='yes'
+  else:
+     arch = 'dskx'
+     commThread ='yes'
+
+  plasticity='no'
+  #logLevel                    = 'warning'
+  logLevel                    = 'warning'
+  logLevel0                   = 'info'
+
+
+5. Submission file for SeisSol on NG:
+
+::
+
+  #!/bin/bash
+  # Job Name and Files (also --job-name)
+
+  #SBATCH -J <job name>
+  #Output and error (also --output, --error):
+  #SBATCH -o ./%j.%x.out
+  #SBATCH -e ./%j.%x.err
+
+  #Initial working directory (also --chdir):
+  #SBATCH --workdir=<work directory>
+
+  #Notification and type
+  #SBATCH --mail-type=END
+  #SBATCH --mail-user=<your email address>
+
+  # Wall clock limit:
+  #SBATCH --time=03:00:00
+  #SBATCH --no-requeue
+
+  #Setup of execution environment
+  #SBATCH --export=NONE
+  #SBATCH --account=<project id>
+  #constraints are optional
+  #--constraint="scratch&work"
+  #SBATCH --partition=general
+
+  #Number of nodes and MPI tasks per node:
+  #max33 so far, else error
+  #SBATCH --nodes=20
+  #SBATCH --ntasks-per-node=1
+  #SBATCH --cpus-per-task=96
+  #Needs specific MPI
+  #module switch mpi.intel mpi.intel/2019
```

Comment:
The line `module load slurm_setup` is important on NG, otherwise stuff might break (according to LRZ)

Reply:
Works for me without slurm_setup

Reply:
The official quote from LRZ is 

> Due to the fact that SLURM performs settings of certain variables in its prologue that can cause MPI crashes, and cannot be overridden in the default module mechanism, the following module should be loaded in SLURM scripts before executing parallel programs
module load slurm_setup

I don't know what it actually does though...

Reply:
ok I ve added that, but anyway it was already loaded before, see

'module load scons cmake/3.6 python/2.7_intel slurm_setup'